extensions/amp-consent/0.1/consent-ui.js

```diff
@@ -232,24 +274,52 @@ export class ConsentUI {
    */
   showIframe_() {
     const {classList} = this.parent_;
+    classList.add(consentUiClasses.iframeActive);
     toggle(dev().assertElement(this.placeholder_), false);
     toggle(dev().assertElement(this.ui_), true);
-    classList.remove('loading');
-    classList.add('i-amphtml-consent-ui-in');
-    classList.add('consent-iframe-active');
+
+    /**
+     * Waiting for mutation twice here.
+     * First mutation is for when the correct elements,
+     * are shown/hidden, and the iframe active class
+     * pushes it out of view.
+     * Second, is for the loading class to be removed.
+     * This will avoid race conditions with the slidein transition.
+     */
+    this.baseInstance_.mutateElement(() => {
+      classList.remove('loading');
+      this.baseInstance_.mutateElement(() => {
+        classList.add(consentUiClasses.uiIn);
+      });
+    });
   }
 
   /**
    * Remove the iframe from doc
    * Remove event listener
    * Reset UI state
+   * Takes in a function to call after our transition has ended
    */
   resetIframe_() {
     const {classList} = this.parent_;
-    classList.remove('i-amphtml-consent-ui-in');
+
+    // Remove the iframe active to go back to our normal height
     classList.remove('consent-iframe-active');
-    this.win_.removeEventListener('message', this.boundHandleIframeMessages_);
-    removeElement(dev().assertElement(this.ui_));
+
+    if (this.isFullscreen_) {
+      this.win_.removeEventListener('message', this.boundHandleIframeMessages_);
```

Comment:
Can we not do if/else, but always set `this.isFullscreen_` to false, remove classLists, eventListener and elements.